tls/s2n_cipher_suites.c

```diff
@@ -1088,7 +1088,7 @@ int s2n_cipher_suites_cleanup(void)
     return 0;
 }
 
-S2N_RESULT s2n_cipher_suite_from_iana(const uint8_t iana[], struct s2n_cipher_suite **cipher_suite)
+S2N_RESULT s2n_cipher_suite_from_iana(const uint8_t iana[S2N_TLS_CIPHER_SUITE_LEN], struct s2n_cipher_suite **cipher_suite)
```

Comment:
Reading the release notes, this is a nice GCC change. They're actually finally going to enforce sizes. "The warning also enables the detection of the likely out of bounds accesses in calls to such functions with smaller arrays."

You can get more compilers (like clang) to enforce the size if you write this as `const uint8_t iana[static S2N_TLS_CIPHER_SUITE_LEN]`. But even then, the compiler will still accept an array >S2N_TLS_CIPHER_SUITE_LEN. See https://godbolt.org/z/M6ch945z1

Reply:
Added the static keyword